### PR TITLE
[Merged by Bors] - Alien cake addict: Allow holding movement keys

### PR DESCRIPTION
**Pull request by RedlineTriad**
I wanted to try one of the new examples but it felt so clunky that I wanted to improve it.

It did make me feel like maybe some input axes abstraction like Unity has might be useful.

Also, eating cake should probably be a separate system from movement.

### CODE REVIEW

**Comment by mockersf** [comment]
please no WASD 😞

**Comment by RedlineTriad** [comment]
> please no WASD disappointed

Why?

**Comment by mockersf** [comment]
Because it doesn't make any sense on anything but QWERTY.

**Comment by RedlineTriad** [comment]
Also works for QWERTZ, but until we have a position based keyboard input system this is probably the best we can do.

**Comment by mockersf** [comment]
because it's not the best we can do, supporting wasd makes a to of games unplayable without changing layout and it's not something I would like to promote in the example. best we can do as arrows

**Comment by RedlineTriad** [comment]
> because it's not the best we can do, supporting wasd makes a lot of games unplayable without changing layout and it's not something I would like to promote in the examples. best we can do is arrows

I feel like it will be hard to do until Bevy has something like https://docs.unity3d.com/Packages/com.unity.inputsystem@1.1/manual/Keyboard.html
> The location of individual keys is agnostic to keyboard layout. This means that, for example, the A key is always the key to the right of the Caps Lock key, regardless of where the currently active keyboard layout places the key that generates an a character, or whether or not the layout doesn't have a key assigned to that character.

But for me `WASD` is also much more comfortable because it's where I normally place my left hand.

Also is this meant as PR feedback or just as a discussion?

**Comment by mockersf** [comment]
> Also is this meant as PR feedback or just as a discussion?

Both: I would really like to not have this in the example, but I can understand it makes sense for part of the world. I also hold no special power to stop it, so it's just my preference. And it's not on a technical point, so I won't put it as a review asking for change.

Just a very strong opinion and preference on my side

**Comment by RedlineTriad** [comment]
I feel like it's worth it since it makes the game nicer for people with `QWERTY`/`QWERTZ` and doesn't really hurt people who have a different layout since arrows are still available. And I don't feel bad about having this code in the examples (where it might be copied from) because we can't really tell developers not to use `WASD` for movement because Bevy doesn't have an alternative I know of and for movement it's the de-facto standard.

**Comment by Dispersia** [comment]
I was interested so googled it, and it appears the french keyboard is the only standardized keyboard that isn't qwerty/qwertz.

 Regardless of that fact, my 60% keyboard doesn't have arrow keys, so that isn't standardized either. If those keys aren't being used for anything else in the demonstration, and the fact that 99% of keyboards would match the schema, I, personally at least, see no harm in it being added.

**Comment by mockersf** [comment]
> the french keyboard is the only standardized keyboard that isn't qwerty/qwertz.

Then it makes sense, I'll go back to my French corner with my French keyboard...

File: examples/game/alien_cake_addict.rs [hunk 1]
```diff
@@ -191,49 +194,61 @@ fn move_player(
...
     mut game: ResMut<Game>,
     mut transforms: Query<&mut Transform>,
+    time: Res<Time>,
 ) {
-    let mut moved = false;
-    let mut rotation = 0.0;
-    if keyboard_input.just_pressed(KeyCode::Up) {
-        if game.player.i < BOARD_SIZE_I - 1 {
-            game.player.i += 1;
+    game.player.move_cooldown = f32::max(game.player.move_cooldown - time.delta_seconds(), 0.);
```
Comment (MinerSebas) [inline examples/game/alien_cake_addict.rs:199]: Why do this manually instead of using a Timer?

Comment (RedlineTriad) [reply examples/game/alien_cake_addict.rs:199]: Good point, I tested it and it seems I can get the same behavior just using a `Timer`.

**Review by MinerSebas — APPROVED** [review]

**Comment by cart** [comment]
@mockersf your "french keyboard perspective" would be valuable in this conversation: #2052.

I'm particularly curious about they physical positions of "scan codes" on french keyboards.

WASD KeyCodes don't just alienate french people. They also alienate "alternative keyboard layout users" like dvorak and colemak. I think we should resolve the "physical key position" conversation in #2052 before making a call on WASD KeyCodes here. Using a pattern in an example is akin to endorsing that pattern and we should encourage people to do the "right thing" here (whatever that ends up being).

**Comment by cart** [comment]
(of course supporting re-mapping is a critical part of the "final" solution)

**Comment by RedlineTriad** [comment]
@mockersf Should I just switch back to arrows? I don't see any progress being made on #2052 so I think it's better to close this PR one way or another.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors try

**Comment by alice-i-cecile** [comment]
Tested locally; this feels much better and is now just uncontroversial QoL and code quality fixes.

**Pull request closed without merging**